File: DKR_1/DKR_1_tabl.py
import math
print('x        y')
print('~~~~~~~~~~~~~~~~~~~~')
for i in range(-40, 36):
    x = i / 5
    if -8 <= x <= 0:
        print(f'{x:6.1f}      не определено')
        continue

    if x < -6:
        y = math.pow(x, 3) * math.pow(x, (0.1 * x)) - math.cos(x)
    elif x < -5:
        y = math.pow(math.exp(x), x) - (math.sin(x) / math.cos(x))
    elif x < 5:
        y = math.pow(x, (0.1 * x)) / 45
    else:
        y = x - math.exp(x * x) / (math.sin(x) / math.cos(x))

    print(f'{x:6.1f}      {y:0.2f}')

print('~~~~~~~~~~~~~~~~~~~~')

File: DKR_1/DKR_1_tabl.pas
uses CRT;
var 
  x, y: real;
  i: integer;
begin
  writeln('x        y');
  writeln('~~~~~~~~~~~~~~~~~~~~');
  
  for i := -40 to 35 do
  begin
    x := i / 5;
    if (x >= -8) and (x <= 0) then
    begin
      writeln(x:6:1, '      не определено');
      continue;
    end;
    if x < -6 then
      y := power(x, 3) * power(x, (0.1 * x)) - cos(x)
    else if x < -5 then
      y := power(exp(x), x) - (sin(x) / cos(x))
    else if x < 5 then
      y := power(x, (0.1 * x)) / 45
    else
      y := x - exp(x*x) / (sin(x) / cos(x));
    
    writeln(x:6:1, '      ', y:0:2);
  end;
  
  writeln('~~~~~~~~~~~~~~~~~~~~');      
end.